### PR TITLE
Model browsing

### PR DESCRIPTION
**Pull request by shshdhdhdhdyx**
Biomodel browsing:

Users can right click anywhere on the screen to see an option called "Browse Biomodels", which will show a search box for users to enter their desired query and search for biomodel(s). They then get a list of results that display the models most closely related to their query, or a single model if they enter a specific model number. They can then click on the desired model, which will pop up in a separate tab as a temp file, converted to Antimony and with all the features available in the extension. They can then modify the model as needed and save it in SBML or Antimony, or close out of it without saving, according to their needs.

### CODE REVIEW

File: vscode-antimony/src/server/stibium/stibium/antimony.lark [hunk 1]
```diff
@@ -76,6 +77,8 @@ ANNOT_KEYWORD: "identity"
     | "isPropertyOf"
     | "hasTaxon"
     | "taxon"
+    | "model_entity_is"
+    | "origin"
```
Comment (longxf21) [inline vscode-antimony/src/server/stibium/stibium/antimony.lark:81]: Highlight these in src/syntax/antimony.tmLanguage.json

File: vscode-antimony/src/server/main.py [hunk 2]
```diff
@@ -125,10 +128,12 @@ def sbml_str_to_ant_str(ls: LanguageServer, args):
     antimony.freeAll()
     ant = antimony.loadSBMLString(sbml)
     if ant < 0:
+        vscode_logger.info("SBML conversion failed")
         return {
             'error': 'Antimony -  {}'.format(antimony.getLastError())
         }
     ant_str = antimony.getAntimonyString(None)
+    vscode_logger.info("SBML conversion passed")
```
Comment (longxf21) [inline vscode-antimony/src/server/main.py:136]: Delete loggers

File: vscode-antimony/src/server/main.py [hunk 3]
```diff
@@ -309,6 +314,50 @@ def recommend(ls: LanguageServer, args):
...
+def get_model(ls: LanguageServer, args):
+    model_id = args[0]
+    model_download_url = ("https://www.ebi.ac.uk/biomodels/search/download?models={model}").format(
+        model=model_id
+    )
+    response = requests.get(model_download_url, stream=True)
+    extract = zipfile.ZipFile(io.BytesIO(response.content))
+    data = io.TextIOWrapper(extract.open(extract.namelist()[0]), encoding="utf-8", newline="").read()
+    extract.close()
+    vscode_logger.info(data)
```
Comment (longxf21) [inline vscode-antimony/src/server/main.py:355]: delete logger

**Review by longxf21 — CHANGES_REQUESTED** [review]

File: vscode-antimony/src/server/main.py [hunk 4]
```diff
@@ -125,6 +128,7 @@ def sbml_str_to_ant_str(ls: LanguageServer, args):
     antimony.freeAll()
     ant = antimony.loadSBMLString(sbml)
     if ant < 0:
+        vscode_logger.info("SBML conversion failed")
```
Comment (longxf21) [inline vscode-antimony/src/server/main.py:131]: Delete this logger. If you want to display this to user, use vscode error message

File: vscode-antimony/src/modelBrowse.ts [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+    async function onQueryConfirmed(query: string, input: MultiStepInput) {
+        var xmlName;
+        window.withProgress({
+			location: ProgressLocation.Notification,
+			title: "Grabbing biomodel...",
+			cancellable: true
+		}, (progress, token) => {
+            return commands.executeCommand('antimony.getModel', query).then(async (result) => {
+                console.log("it was successful!")
```
Comment (longxf21) [inline vscode-antimony/src/modelBrowse.ts:71]: Delete this log

File: vscode-antimony/src/modelBrowse.ts [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+        vscode.commands.executeCommand('antimony.sbmlStrToAntStr', xmlData).then(async (result: any) => {
+            const fileName = path.basename(xmlName, ".xml")
+            const tempDir = os.tmpdir()
+            var tempFile = `${fileName}.ant`
+            tempPath = path.join(tempDir, tempFile)
+            fs.writeFile(tempPath, String(result), (error) => {
+                if (error) {
+                    console.log(error)
+                } else {
+                    console.log("Written to file")
```
Comment (longxf21) [inline vscode-antimony/src/modelBrowse.ts:90]: Delete this log, or have a better logging message

File: vscode-antimony/src/modelBrowse.ts [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+                if (result.error) {
+                    this.current.items = [];
+
+                    // Don't display errors too often
+                    const curMillis = new Date().getTime();
+                    if (curMillis - this.lastErrorMillis < 1000) {
+                        return;
+                    }
+                    this.lastErrorMillis = curMillis;
+                    window.showErrorMessage(`Could not perform query: ${result.error}`).then(() => console.log('finished'));
```
Comment (longxf21) [inline vscode-antimony/src/modelBrowse.ts:253]: Why a console log and .then() here?

**Review by longxf21 — CHANGES_REQUESTED** [review]

**Review by evaliu2002 — CHANGES_REQUESTED** [review]
Line 492's indentation off. Otherwise should look good. Check Aldrich's review

**Review by evaliu2002 — APPROVED** [review]

**Pull request merged**